satellite/admin-back-office-v1.md

```diff
@@ -0,0 +1,617 @@
+- Segment limit, and enter the max number with formatted numerical input fields for better
+  readability.
+- Bucket limit, and enter the number of buckets.
+
+__When an account needs to be deleted__
+
+I want authorized users to be able to press a "delete account" button along with a "confirm" delete
+button that confirms this user and all their information and stored data are to be deleted, so that
+we don't have to manually access their API keys and manually delete their data.
+
+__When an account needs to be created__
+
+I want authorized users to have an option in their initial login view to create a user.
+
+This operation rarely happens so the design of the UX shouldn’t make it prominent
+
+Acceptance criteria:
+- Input fields for: full name, email, generate (temporary) password.
+
+__When an account needs more projects__
+
+I want to locate the user I need to enable a higher project allowance for and have an option to
+increase the number of projects the account is allowed to have in the account details view.
+
+Acceptance criteria:
+- Field to increase or decrease the account's project count.
+- Project limits default to the user's project limits or can be set if otherwise.
+
+__When a user needs to be added to a different user's project__
+
+I want to be able to search for the project ID.
+I need to add the user to, so that I can view the members off that project and add and/or remove a
+user.
+
+__When a user needs their multi factor authentication (MFA) reset__
+
+I want there to be a MFA reset option in the account details view.
+
+Acceptance criteria:
+- We can see that MFA is enabled or disabled.
+- We can only disable MFA.
+
+__When a user wants to change the email address tied to their account__
+
+I want to be able to edit the email address in the account details view so that all events and data
+associated to that user is now under the updated email address.
+
+##### Modifications history
+
+The back office keeps a register of all performed operations and who executed them that cause
+modifications.
+
+Each record has the following information:
+- Timestamp. When the operation was performed.
+- Operator email. Who performed the operation.
+- Account ID. What account was modified.
+- Entity that was modified (i.e. account, project, bucket) and ID.
+- Operation. What was done (e.g. change email address, set user agent, etc.). The text used to
+  identify the operation has to be unique inside of the entity and human friendly.
+- Current data. The new data that was modified with this operation. This is a structured blob
+  because it could be more than one modified field per operation.
+- Previous data. The existing data previous to the modification performed by this operation. This is
+  a structured blob because it could be more than one modified field per operation.
+
+This information is stored in the same database that the satellite uses.
+
+The _DBX_ schema is
+
+```
+model admin_operation_modifications_history (
```

Comment:
even if we do not use it at first, I think it would be a good idea to include a "notes" or "metadata" field in this table

That way, in atypical situations, the operator would have the option of adding additional notes about what they did/why. I wouldn't expect to use it in most cases, but I imagine it could be nice in unusual circumstances to be able to add optional meta notes about the operation

Reply:
At this time the back office doesn't have the option to add notes to any of the operations that offers.

Adding a column to the table that can be empty and not reference any other entity is an easy task that can be done through a migration; it will require a lot more work to add that notes/metadata field to every API endpoint and modify the UI that the database change, so I feel that adding the column at this stage is a "premature feature" (a term adopted from "premature optimization" :joy:) and it may confuse people how read the code in the future.

Reply:
ok good point, and I agree.